### PR TITLE
Remove `scan_csv` methods from `LogicalPlanBuilder`

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/2536

 # Rationale for this change

- `LogicalPlanBuilder` should be in same crate as `LogicalPlan`
- `LogicalPlanBuilder` should not need to know about data sources & object store but just `TableSource`

# What changes are included in this PR?

- Move some logic from `LogicalPlanBuilder` to `SessionContext`
- Reduce duplication in the test code
- Ignored one problematic test and filed https://github.com/apache/arrow-datafusion/issues/2546 to fix

There will be separate PRs to remove the other scan methods (parquet, json, avro).

# Are there any user-facing changes?

Yes, this changes the `LogicalPlanBuilder` API

### CODE REVIEW

File: ballista/rust/core/src/serde/logical_plan/mod.rs [hunk 1]
```diff
@@ -1236,39 +1212,17 @@ mod roundtrip_tests {
 
     #[tokio::test]
     async fn roundtrip_analyze() -> Result<()> {
-        let schema = Schema::new(vec![
```
Comment (alamb) [inline ballista/rust/core/src/serde/logical_plan/mod.rs:1239]: nice cleanups

File: ballista/rust/core/src/serde/logical_plan/mod.rs [hunk 2]
```diff
@@ -1402,31 +1293,17 @@ mod roundtrip_tests {
...
+            .await?
+            .aggregate(vec![col("state")], vec![max(col("salary"))])?
+            .build()?;
 
         roundtrip_test!(plan);
 
         Ok(())
     }
 
+    #[ignore] // see https://github.com/apache/arrow-datafusion/issues/2546
```
Comment (alamb) [inline ballista/rust/core/src/serde/logical_plan/mod.rs:1306]: FYI other reviewers: https://github.com/apache/arrow-datafusion/issues/2546

**Review by alamb — APPROVED** [review]
I went through the test changes carefully; Very nice @andygrove

**Comment by andygrove** [comment]
Thanks for the review @alamb ... I am close to being able to move `LogicalPlanBuilder` to the `datafusion-expr` crate now.

**Pull request merged**